src/Analysis/GWASV2/SelectStudyPopulation/SelectStudyPopulation.jsx

```diff
@@ -1,37 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import { Spin } from 'antd';
-import CohortSelect from './Utils/CohortSelect';
-import { useSourceFetch } from '../../GWASWizard/wizardEndpoints/cohortMiddlewareApi';
+import AddCohortButton from './Utils/AddCohortButton';
+import CohortDefinitions from './Utils/CohortDefinitions';
+/* Eslint is giving error: import/no-named-as-default-member: needs a parser plugin */
+/* eslint-disable-next-line */
+import SearchBar from '../Shared/SearchBar';
 import './SelectStudyPopulation.css';
 
 const SelectStudyPopulation = ({
   selectedStudyPopulationCohort,
-  setSelectedStudyPopulationCohort,
-  current,
+  handleSelectStudyPopulation,
+  dispatch,
```

Comment:
it would be good to get rid of unnecessary handlers now that we have `dispatch`

Reply:
The prop handleSelectStudyPopulation is used to call the dispatch function when the CohortSelect element is used in CohortSelect, but in CustomDichotomousCovariates.jsx it is used to: 

setFirstPopulation
setSecondPopulation

I find this a little confusing but don't have a better idea to improve it. 
I have now deleted the extra dispatch prop which gets rid of an unnecessary prop.

Reply:
thanks @jarvisraymond-uchicago . I've attempted to simplify the code here https://github.com/uc-cdis/data-portal/pull/1141/commits/c66c99c90ab778e436493310b5d824fc3fcfcc1e. The idea of this change is to make the CohortSelect component simpler and more generic, focused solely on showing cohorts and passing the selected cohort to the handler. Then the external wrapping/calling component can decide how the handler logic should look like in it own context. Let me know what you think.